### PR TITLE
[SPARK-25842][SQL] Deprecate rangeBetween APIs introduced in SPARK-21608

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?
See the detailed information at https://issues.apache.org/jira/browse/SPARK-25841 on why these APIs should be deprecated and redesigned.

This patch also reverts https://github.com/apache/spark/commit/8acb51f08b448628b65e90af3b268994f9550e45 which applies to 2.4.

## How was this patch tested?
Only deprecation and doc changes.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Yup, I also agree with this revert.

File: python/pyspark/sql/window.py [hunk 1]
```diff
@@ -239,34 +212,27 @@ def rangeBetween(self, start, end):
         and "5" means the five off after the current row.
 
         We recommend users use ``Window.unboundedPreceding``, ``Window.unboundedFollowing``,
-        ``Window.currentRow``, ``pyspark.sql.functions.unboundedPreceding``,
-        ``pyspark.sql.functions.unboundedFollowing`` and ``pyspark.sql.functions.currentRow``
-        to specify special boundary values, rather than using integral values directly.
+        and ``Window.currentRow`` to specify special boundary values, rather than using integral
```
[the remark this answers is not in the document]
Comment (rxin) [reply python/pyspark/sql/window.py]: what do you mean? the old rangeBetween API is still valid.

[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply python/pyspark/sql/window.py]: Oops, right. I misread.

**Comment by HyukjinKwon** [comment]
Looks good to me.

**Review by srowen — COMMENTED** [review]
I don't object, seems low risk to slip into 2.4. I don't know much about this code.

**Comment by cloud-fan** [comment]
thanks, merging to master/2.4!

**Pull request closed without merging**